include/cpuinfo_x86.h

```diff
@@ -81,12 +81,44 @@ typedef struct {
   // Make sure to update X86FeaturesEnum below if you add a field here.
 } X86Features;
 
+// Increase this value if more cache levels are needed.
+#ifndef CPU_FEATURES_MAX_CACHE_LEVEL
+#define CPU_FEATURES_MAX_CACHE_LEVEL 10
+#endif
+
+typedef enum {
+  EMPTY = 0,
+  DATA_CACHE = 1,
+  INSTRUCTION_CACHE = 2,
+  UNIFIED_CACHE = 3,
+  TLB = 4,
+  DTLB = 5,
+  STLB = 6,
+  PREFETCH = 7
+} X86CacheType;
+
+typedef struct {
+  int level;
+  X86CacheType cache_type;
+  int cache_size;  // Cache size in kBytes
```

Comment:
[KiB](https://en.wikipedia.org/wiki/Kibibyte) or [KB](https://en.wikipedia.org/wiki/Kilobyte) ?
It may be best to leave it as bytes then, `int` can store up to 2 GiB.